### PR TITLE
Move more cross model secret facade logic to use the service service

### PR DESCRIPTION
**Pull request by wallyworld**
The remaining logic in the various cross model facades which access secret info from state is ported to use the secret service. The service just has no op methods for now,

Driveby - rename a params SecretRevisionChange Revision attr to LatestRevision to make its use more clear.

Also, add a table to record a remote cross model secret latest revision. The consuming model needs to know the latest revision of a cross model secret. For local secrets, this can be inferred, but for cross model, we need to record it.

From secret remote consumers. we don't record the unit uuid (unit doesn't exist in the model). We record the unit name, so update the table.

## QA steps

Just unit tests for now
Double check - bootstrap a controller and add a secret to the controller charm

## Links

**Jira card:** JUJU-5875

### CODE REVIEW

File: domain/schema/secret.go [hunk 1]
```diff
@@ -273,19 +286,15 @@ CREATE UNIQUE INDEX idx_secret_unit_consumer_label ON secret_unit_consumer (labe
 
 CREATE TABLE
     secret_remote_unit_consumer (
-        uuid TEXT PRIMARY KEY,
         secret_id TEXT NOT NULL,
-        unit_uuid TEXT NOT NULL,
+        unit_id TEXT NOT NULL,
```
Comment (ycliuhw) [inline domain/schema/secret.go:290]: A comment like line 127 could help to understand why it's not a FK anymore and also mention it's on the offering model side.
I didn't have proper comment, could you help to add a comment to mention the purpose of creating this table? Thanks.

**Review by ycliuhw — APPROVED** [review]
Thank you for the change!

File: domain/schema/secret.go (hunk 1 above)
Comment (wallyworld) [reply domain/schema/secret.go:290]: Added comments

**Pull request merged**